s3/dispatch.c

```diff
@@ -284,7 +285,11 @@ callback_response(void * cookie, struct http_response * res)
 			/* We have no Content-Length header. */
 			clen = (uint32_t)(-1);
 		} else {
-			clen = strtoull(s_clen, NULL, 0);
+			/* Parse in base 10, no trailing characters. */
+			if (PARSENUM_EX(&clen, s_clen, 10, 0)) {
+				warnp("Invalid Content-Length: %s", s_clen);
+				goto err1;
```

Comment:
A bogus response from a web server is not an *internal* error.  I'd replace `goto err1;` with `clen = (uint32_t)(-1);` indicating that there is no (valid) Content-Length header.